Mapbox/MapboxMapsFoundation/Camera/CameraManager.swift

```diff
@@ -642,6 +644,36 @@ public class CameraManager {
         animationGroup.delegate = animationGroup
         cameraLayer.add(animationGroup, forKey: animationKey)
     }
+
+    /**
+    This function optimizes the bearing for set camera so that it is taking the shortest path
+
+    - Parameter startBearting: The current or start bearing of the map viewport
+    - Parameter endBearting: The bearing of where the map viewport should end at
+    - Returns: A `CLLocationDirection` that represents the correct final bearing accounting for positive and negatives
+    */
+    internal func optimizeBearing(startBearing: CLLocationDirection?, endBearing: CLLocationDirection?) -> CLLocationDirection? {
+        // This modulus is required to account for larger values
+        guard
+            let startBearing = startBearing?.truncatingRemainder(dividingBy: 360.0),
```

Comment:
This is interesting.. Can `startBearing` ever be greater than ±360? If so, then the camera animation will still result in a "spin". We're not changing the `startBearing` before calling `setCamera` on line 234 above.

Reply:
In an optimized setting no! for example, if I am doing setCamera for the first time and pass like 500 or -500 to start off. It will go through the optimize bearing and result in  a -360 < bearing < 360 situation. This pattern will continue

Reply:
Right, it's clear that if we are restricted to the "optimized" flow, there would be no problem.. But is it possible to bypass this somewhere else which could result in this logic breaking.

Reply:
tested this and confirmed that it does not break